tests/db_functions/test_dates.py

```diff
@@ -0,0 +1,611 @@
+from __future__ import unicode_literals
+
+from datetime import datetime
+from unittest import skipIf
+
+from django.conf import settings
+from django.db import connection
+from django.db.models import DateField, DateTimeField
+from django.db.models.functions.date import (
+    Extract, ExtractDay, ExtractHour, ExtractMinute, ExtractMonth,
+    ExtractSecond, ExtractWeekDay, ExtractYear, Trunc, TruncDay, TruncHour,
+    TruncMinute, TruncMonth, TruncSecond, TruncYear,
+)
+from django.test import TestCase, override_settings
+from django.utils import timezone
+
+from .models import DTModel
+
+try:
+    import pytz
+except ImportError:
+    pytz = None
+
+
+def microsecond_support(value):
+    return value if connection.features.supports_microsecond_precision else value.replace(microsecond=0)
+
+
+def truncate_to(value, kind, tzinfo=None):
+    def truncate(value, kind):
+        if kind == 'second':
+            return value.replace(microsecond=0)
+        if kind == 'minute':
+            return value.replace(second=0, microsecond=0)
+        if kind == 'hour':
+            return value.replace(minute=0, second=0, microsecond=0)
```

Comment:
I see two potential problems here:

1) If `value` is aware, this requires normalization in case the change makes you hit the DST threshold.

2) The correct semantic for the database is to operate in the current time zone. If this function is expected to do the same thing, the current implementation is insufficient, because some time zone offsets aren't integer number of hours.

I'm not sure what this function is used for, so perhaps this doesn't matter.